src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -2292,6 +2298,44 @@ int SketchObject::trim(int GeoId, const Base::Vector3d& point)
         Geometry.setValues(geomlist);
     };
 
+    auto getTransformParamsAndDeleteConstraints = [this, isPointAtPosition, handlemultipleintersection]
+        (int GeoId, int GeoId1, Base::Vector3d point1,  ConstraintType &constrType1, PointPos &secondPos1) {
+        const std::vector<Constraint *> &constraints = this->Constraints.getValues();
+        int constrId=0;
+        std::vector<int> delete_list;
+        for (std::vector<Constraint *>::const_iterator it=constraints.begin(); it != constraints.end(); ++it) {
+            Constraint *constr = *(it);
+            if (constr->Type == Sketcher::PointOnObject && constr->First == GeoId1  && constr->Second == GeoId) {
+                if (isPointAtPosition(GeoId1, constr->FirstPos, point1)) {
+                    constrType1 = Sketcher::Coincident;
+                    secondPos1 = constr->FirstPos;
+                    handlemultipleintersection(constr, GeoId, start, secondPos1);
+                    delete_list.push_back(constrId);
+                }
+            }
+            constrId++;
+        }
+        constrId=0;
+        for (std::vector<Constraint *>::const_iterator it=constraints.begin(); it != constraints.end(); ++it) {

```

Comment:
It looks to me that this for loop may be merged with the for loop above (single loop with two if inside its body). Am I wrong?

Reply:
I did this for a reason (initially I incuded everything in one single loop)

It is possible that the Sketch has both PointOnObject constrait to the intersecting line and a Simple Tangent. In this case we want to change to a single end-to-end tangency, i.e we want to ensure that constrType1 is set to Sketcher::Tangent.
The double loop ensures that the ordering of the constraints does not influence the result. 

Of course a double loop is more expensive than a single loop. But as this only happens when trimming and it is very quick. I don't think it is worth figuring out how to merge the loops.

Reply:
I understand. Please document this design choice in the code in a comment. If you do not, somebody is going to see that code and refactor it into a single loop leading to unwanted behaviour.

Reply:
Ok, done. I also rebased and squashed the commits into one nice commit.